Core/Src/main_task.c

```diff
@@ -604,7 +721,11 @@ static void run_downshift_sm(void)
 		// lever does not seem to be as important for downshifts, but still
 		// give some time for it
 		begin_shift_tick = HAL_GetTick();
-		initial_gear = get_current_gear();
+
+#ifdef NO_GEAR_POT
+		tcm_data.current_gear = get_current_gear();
```

Comment:
Why is there any difference between NO_GEAR_POT and gear pot here?

Reply:
Because you said itd be unnecessary to check the gear again here with the gear pot updating it every loop, but I thought if we're doing the transmission speed based one we need to find the latest value?